pages/collections/index.tsx

```diff
@@ -1,14 +1,20 @@
+import { useQuery } from '@apollo/client'
 import Collection from '../../components/Collection'
-import { CollectionsProps } from '../../components/helpers/interfaces'
-import { getCollectionList } from '../../services/reventService'
+import { GET_COLLECTIONS } from '../../lib/graphql'
+
+export default function Collections() {
+  const { loading, error, data } = useQuery(GET_COLLECTIONS)
+
+  const collectionListing = data?.collections
 
-export default function Collections({ collectionList }: CollectionsProps) {
   return (
     <div className='p-8 bg-gray-100'>
       <div className='px-8'>
         <main className='min-h-screen py-8 px-20 flex-1 flex flex-col'>
+          {loading && <p>LOADING</p>}
+          {error && <p>Error while retrieving the collections</p>}
           <div className='grid grid-cols-4 gap-4'>
-            {collectionList.map((collection) => (
+            {collectionListing?.map((collection) => (
               <Collection collection={collection} key={collection.id} />
             ))}
           </div>
```

Comment:
with this you can also replace `collectionListing?` with `data.collections` because there will always be a data variable so you don't need the `?`

Reply:
no, it always displays the list. It might seem like it doesn't because initially the list is empty, but the div is always there, it should be rendered only if not loading and not error